src/app/search-page/search-page.component.ts

```diff
@@ -0,0 +1,33 @@
+import {Component, OnDestroy, OnInit} from '@angular/core';
+import { CocktailsService } from '../../services/cocktails.service';
+import {Subscription} from 'rxjs/Subscription';
+
+@Component({
+  selector: 'app-search-page',
+  templateUrl: './search-page.component.html',
+  styleUrls: ['./search-page.component.css']
+})
+export class SearchPageComponent implements OnInit, OnDestroy {
+  cocktailsList: Array<Object> = [];
+  filteredCocktailsList: Array<Object> = [];
+  cocktailsSubscription: Subscription;
+  constructor(private cocktailsService: CocktailsService) { }
+
+  ngOnInit () {
+    const cocktailsSource$ = this.cocktailsService.getCocktails();
```

Comment:
is there any reason for "$" symbol in the variable name?

Reply:
As Sergiy mentioned, it's a naming convention for saving observable into variable

Reply:
ok